release_management/create_artifacts.py

```diff
@@ -90,15 +95,32 @@ def _stratisd_artifacts(namespace):
     if release_version != namespace.version:
         raise RuntimeError("Version mismatch.")
 
-    make_source_tarball("stratisd", release_version, output_abs_path)
-
-    vendor_tarfile_name = vendor(manifest_abs_path, release_version)
-
-    os.rename(vendor_tarfile_name, os.path.join(output_abs_path, vendor_tarfile_name))
-
-    crate_name = f"stratisd-{release_version}.crate"
-    crate_path = os.path.join("target", "package", crate_name)
-    os.rename(crate_path, os.path.join(output_abs_path, crate_name))
+    if namespace.pre_release_suffix:
+        print(f"Using suffix: {namespace.pre_release_suffix}")
+        release_suffix_version = release_version + namespace.pre_release_suffix
+        make_source_tarball("stratisd", release_suffix_version, output_abs_path)
+        vendor_tarfile_name = vendor(
+            manifest_abs_path, release_version, suffix=namespace.pre_release_suffix
+        )
+        os.rename(
+            vendor_tarfile_name, os.path.join(output_abs_path, vendor_tarfile_name)
+        )
+        crate_name = f"stratisd-{release_version}.crate"
+        crate_path = os.path.join("target", "package", crate_name)
+        crate_suffix_name = f"stratisd-{release_suffix_version.replace('~','-')}.crate"
```

Comment:
What entity requires you to replace ```~``` with ```-```? Typical suffix, e.g., "20220628git0a18d116.el9" contains no ```~```.

Reply:
It's the `%{crates_source}` RPM macro.  It has a lua script with this:
```
version = version:gsub('~', '-')
```
I'm guessing it's to ensure that there is no `~` character in the filename, to ensure that it is not present in the crates.io URL.  If I don't do that, mock will not be able to find the source file.

For a build test today, the generate_sourcerpms.sh script outputs these files:
```
SOURCES/stratisd-3.3.0~20220808git1fa6d536.tar.gz
SOURCES/stratisd-3.3.0-20220808git1fa6d536.crate
SOURCES/stratisd-3.3.0~20220808git1fa6d536-vendor.tar.gz
```

Reply:
And Fedora requires the '~'.